packages/odyssey-react/src/components/Box/Box.tsx

```diff
@@ -11,128 +11,501 @@
  */
 
 import React, { forwardRef } from "react";
-import type { ComponentPropsWithRef } from "react";
-import { useCx, useOmit, withStyles } from "../../utils";
+import { useCx, useOmit, withStyles, PolymorphicForwardRef } from "../../utils";
 import styles from "./Box.module.scss";
 
 type spacing = "xs" | "s" | "m" | "l" | "xl";
+type overflow = "visible" | "hidden" | "clip" | "scroll" | "auto";
+type borderRadius = "base" | "outer" | "none";
 
-export interface BoxProps
-  extends Omit<ComponentPropsWithRef<"div">, "style" | "className"> {
+export interface BoxProps {
   /**
-   * Padding applied inside box
+   * Display type
    */
-  padding?: spacing;
+  display?:
+    | "block"
+    | "inline"
+    | "inline-block"
+    | "flex"
+    | "inline-flex"
+    | "grid"
+    | "inline-grid"
+    | "none";
   /**
-   * Border colors
+   * Positioning
    */
-  borderColor?: "display" | "none";
+  position?: "absolute" | "relative" | "static" | "fixed" | "sticky";
   /**
-   * Hover border
+   * Grid columns
    */
-  hoverBorderColor?: "interactive" | "none";
+  gridTemplateColumns?: "2" | "3" | "2-auto-1" | "3-auto-1-auto-3";
   /**
-   * Border radius
+   * Grid columns
    */
-  borderRadius?: "base" | "none";
+  gridColumn?: "1" | "2" | "3";
   /**
-   * Shadow
+   * Grid rows
    */
-  boxShadow?: "default" | "none";
+  gridRow?: "1" | "2" | "3" | "4" | "5";
   /**
-   * Hover shadow
+   * [Main axis](https://developer.mozilla.org/en-US/docs/Glossary/Main_Axis) of flex layout
    */
-  hoverBoxShadow?: "default" | "none";
+  flexDirection?: "row" | "row-reverse" | "column" | "column-reverse";
   /**
-   * Focus indicator
+   * Whether flex items are forced onto one line or can wrap
    */
-  focusRing?: "primary" | "none";
+  flexWrap?: "nowrap" | "wrap" | "wrap-reverse";
   /**
-   * Display type
+   * Flex controls how a flex item both grows and shrinks.
+   *
+   * - Use `initial` to allow a flex item to shrink but not grow, taking into account its initial size
+   * - Use `auto` to allow a flex item to grow and shrink, taking into account its initial size
+   * - Use `evenly` to allow a flex item to grow and shrink as needed, ignoring its initial size
+   * - Use `none` to prevent a flex item from growing or shrinking
    */
-  display?: "flex" | "none";
+  flex?: "initial" | "auto" | "evenly" | "none";
```

Comment:
for these optional types if would be good to specify the default using JSDOC `@default` annotation in the JSDOC comment so it is picked up by our react docgen tooling

Reply:
We don't assign a default value through the API, specifying a default would suggest that we do.